generators/entity-client/templates/angular/src/main/webapp/app/entities/_entity-management.component.ts

```diff
@@ -88,26 +88,26 @@ export class <%= entityAngularName %>Component implements OnInit, OnDestroy {
 
         <%_ } _%>
         <%_ if (pagination === 'pagination' || pagination === 'pager') { _%>
-    private onSuccess(data, headers) {
+    private onSuccess(data: <%= entityAngularName %>[], headers: HttpHeaders) {
         <%_ if (databaseType !== 'cassandra') { _%>
         this.links = this.parseLinks.parse(headers.get('link'));
-        this.totalItems = headers.get('X-Total-Count');
+        this.totalItems = parseInt(headers.get('X-Total-Count'), 10);
         this.queryCount = this.totalItems;
         // this.page = pagingParams.page;
         <%_ } _%>
         this.<%= entityInstancePlural %> = data;
     }
         <%_ } else if (pagination === 'infinite-scroll') { _%>
-    private onSuccess(data, headers) {
+    private onSuccess(data: <%= entityAngularName %>[], headers: HttpHeaders) {
         this.links = this.parseLinks.parse(headers.get('link'));
-        this.totalItems = headers.get('X-Total-Count');
+        this.totalItems = parseInt(headers.get('X-Total-Count'), 10);
         for (let i = 0; i < data.length; i++) {
             this.<%= entityInstancePlural %>.push(data[i]);
         }
     }
 
     <%_ }} _%>
-    private onError(error) {
-        this.jhiAlertService.error(error.message, null, null);
+    private onError(errorMessage: string) {
```

Comment:
So are we getting message directly without an error object?

Reply:
Please look it up in `pagination-template.ejs` or `no-pagination-template.ejs` or `infinite-scroll-template.ejs`. In all files, `onError` is called with `this.onError(res.message)`, so the error object is already navigated to the message.

We can also change the other templates instead if you prefer to have the error object in the handler but my solution would conform to the success handler which is called like this `this.onSuccess(res.body, res.headers)`.

I have not checked if the current state actually works / shows errors. Maybe it is the global error handler jumping in and this one isn't needed at all?

Reply:
Its ok, most probably I wrote those but it has been a while. Just asking to confirm.